release-22.2: sql: add tests for CTAS, CMVAS with every SHOW statement

ecwall opened the pull request:
Backport 2/2 commits from #106407.

/cc @cockroachdb/release

---

Fixes #105895

Adds tests for CREATE TABLE AS, CREATE MATERIALIZED VIEW AS sourcing from
all SHOW statements.

Release note: None

Release justification: Add test coverage.

ecwall committed:
sql: fix ShowSurvivalGoal comment

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107166)
<!-- Reviewable:end -->

chengxiong-ruan commented on pkg/sql/create_as_test.go:
```diff
@@ -96,6 +96,207 @@ func TestCreateAsVTable(t *testing.T) {
+				i, testCase.sql,
+			)
+			sqlRunner.Exec(t, createTableStmt)
+			// Skip `CREATE MATERIALIZED VIEW` in 22.2 to avoid
```
How about doing `SELECT @1` from them?

ecwall committed:
sql: add tests for CTAS, CMVAS with every SHOW statement

Fixes #105895

Adds tests for CREATE TABLE AS, CREATE MATERIALIZED VIEW AS sourcing from
all SHOW statements.

Release note: None
